### PR TITLE
code added for nodeJS-week2 homework

### PR DESCRIPTION
**Pull request by rismita87**
nodeJS-week2 homework

### CODE REVIEW

File: nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    throw error;
+  }
+  //Get meals that has a price smaller than maxPrice
+  const maximumPrice = request.query.maxPrice;
+  const mealTitle = request.query.title;
+  const createdAfter = request.query.createdAfter;
+  const limit = request.query.limit;
+  let keysAreInValid = false;
+  //If the users writes a query parameter that is not supported
+  Object.keys(request.query).forEach(function (key) {
```
Comment (jslambda) [inline nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js]: This check could have been written more concisely, something like:
const parameterKeys = Object.keys(request.query).filter(...)
....

File: nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    }
+  } else if (limit) {
+    if (isNaN(limit)) {
+      //respond with status 400 when the limit is not parseable
+      response.status(400).send("wrong limit format");
+    } else {
+      const limitedMeal = meals.slice(0, limit); //return only specific number of meals
+      response.send(limitedMeal);
+    }
+  } else {
```
Comment (jslambda) [inline nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js:85]: Two more points:
1. Can you rewrite the code so it takes all the valid parameters into account (there can be more than one valid parameter in the request)?
2. Can you break down this code into smaller functions?

File: nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+//Respond with the json for the meal with the corresponding id
+router.get("/:id", async (request, response) => {
+  const id = request.params.id;
+  if (isNaN(id)) {
+    //respond with status 400 when  is not parseable
+    response.status(400).send("wrong id format");
+  } else if (id > meals.length) {
+    response.send("No meal with this id number!!");
+  } else {
+    response.send(meals[id - 1]);
```
Comment (jslambda) [inline nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js]: I don't think this is the correct implementation. You need to find the with the id.

File: nodejs/week2/homework/meal-sharing/src/backend/api/reservations-router.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+router.get("/", async (request, response) =>
+  //response.send({ reservations }) //Respond with the json for all reservations but the test is failing
+  response.send(reservations)
+);
+
+//Respond with the json for the reservation with the corresponding id *
+router.get("/:id", async (request, response) => {
+  const id = request.params.id;
+  if (id) {
+    response.send(reservations[id - 1]);
```
Comment (jslambda) [inline nodejs/week2/homework/meal-sharing/src/backend/api/reservations-router.js]: The same problem as https://github.com/rismita87/hyf-homework/pull/18/files#r846661591

**Comment by rismita87** [comment]
Thanks for your feedback. I will make the changes accordingly.
regards
Rismita

**Pull request merged**